superset/charts/schemas.py

```diff
@@ -59,3 +60,460 @@ class ChartPutSchema(Schema):
+        "which are defined only within the query object. See "
+        "`ChartDataAdhocMetricSchema` for the structure of ad-hoc metrics.",
+    )
+    post_processing = fields.List(
+        fields.Nested(ChartDataPostProcessingOperationSchema),
+        description="Post processing operations to be applied to the result set. "
+        "Operations are applied to the result set in sequential order.",
+        required=False,
+    )
+    time_range = fields.String(
+        description="A time rage, either expressed as a colon separated string "
+        "`since : until`. Valid formats for `since` and `until` are: \n"
+        "- ISO 8601\n"
+        "- X days/years/hours/day/year/weeks\n"
+        "- X days/years/hours/day/year/weeks ago\n"
+        "- X days/years/hours/day/year/weeks from now\n"
+        "\n"
+        "Additionally, the following freeform can be used:\n"
+        "\n"
+        "- Last day\n"
+        "- Last week\n"
+        "- Last month\n"
+        "- Last quarter\n"
+        "- Last year\n"
+        "- No filter\n"
+        "- Last X seconds/minutes/hours/days/weeks/months/years\n"
+        "- Next X seconds/minutes/hours/days/weeks/months/years\n",
+        required=False,
+        example="Last week",
+    )
+    time_shift = fields.String(
+        description="A human-readable date/time string. "
+        "Please refer to [parsdatetime](https://github.com/bear/parsedatetime) "
+        "documentation for details on valid values.",
+        required=False,
+    )
+    is_timeseries = fields.Boolean(
+        description="Is the `query_object` a timeseries.", required=False
+    )
+    timeseries_limit = fields.Integer(
+        description="Maximum row count for timeseries queries. Default: `0`",
+        required=False,
+    )
+    row_limit = fields.Integer(
+        description='Maximum row count. Default: `config["ROW_LIMIT"]`', required=False,
+    )
+    order_desc = fields.Boolean(
+        description="Reverse order. Default: `false`", required=False
+    )
+    extras = fields.Dict(description=" Default: `{}`", required=False)
+    columns = fields.List(fields.String(), description="", required=False,)
+    orderby = fields.List(
+        fields.List(fields.Raw()),
+        description="Expects a list of lists where the first element is the column "
+        "name which to sort by, and the second element is a boolean ",
+        required=False,
+        example=[["my_col_1", False], ["my_col_2", True]],
+    )
+
+
+class ChartDataDatasourceSchema(Schema):
+    description = "Chart datasource"
+    id = fields.Integer(description="Datasource id", required=True,)
+    type = fields.String(description="Datasource type", enum=["druid", "sql"])
+
+
+class ChartDataQueryContextSchema(Schema):
+    datasource = fields.Nested(ChartDataDatasourceSchema)
+    queries = fields.List(fields.Nested(ChartDataQueryObjectSchema))
+
+    # pylint: disable=no-self-use
```

Comment:
nice way of doing it also

Reply:
I wasn't able to ignore this any other way. It seems black isn't able to place the ignore string on the right row.